### ISSUE TITLE
AcceptV6BoundToAnyV6_Success failed on OS X in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#4890]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/osx_release_tst_prtest/66/console
```
12:50:37    System.Net.Sockets.Tests.DualMode.AcceptV6BoundToAnyV6_Success [FAIL]
12:50:37       System.ArgumentException : The supplied System.Net.Internals.SocketAddress is an invalid size for the System.Net.IPEndPoint end point.
12:50:37       Parameter name: socketAddress
12:50:37       Stack Trace:
12:50:37             at System.Net.Sockets.IPEndPointExtensions.Create(EndPoint thisObj, SocketAddress socketAddress)
12:50:37             at System.Net.Sockets.Socket.Accept()
12:50:37             at System.Net.Sockets.Tests.DualMode.Accept_Helper(IPAddress listenOn, IPAddress connectTo)
```

**Comment by eerhardt** [comment]
I've run this tests 20+ times on my mac, and I haven't seen it fail.

I'm thinking of closing this issue until we see it happen again.  Thoughts, @stephentoub @ericeil @pgavlin ?

**Comment by stephentoub** [comment]
@eerhardt, just to confirm, you're using the one in System.Net.Sockets.APMServer.Tests?  It looks like a version of the test with the same name is also present in System.Net.Sockets.Tests.

**Comment by eerhardt** [comment]
Yes, the tests I've been running are bin located under `/corefx/bin/                                                            tests/OSX.AnyCPU.Debug/System.Net.Sockets.APMServer.Tests/dnxcore50/`

I've been running them by:

```
/Users/eerhardt/git/corefx/src/System.Net.Sockets.Legacy/tests/FunctionalTests$ msbuild /t:BuildAndTest /p:RestorePackages=false System.Net.Sockets.APMServer.Tests.csproj
```

where 'msbuild' is an alias to running msbuild on mono.

**Comment by stephentoub** [comment]
Thanks.

I don't know under what conditions that particular failure repros.  What concerns me is that it it's not an error indicating a timeout or some other random failure, but would seem to potentially indicate some kind of corruption or race condition or something like that.

Rather than using msbuild each time, you might just build it once and then use corerun and xunit.console.netcore.exe to run that test directly.  In the past when I've wanted to run a test many times I've also done something like this:
```C#
public static IEnumerable<object[]> Iterations
{
    get { for (int i = 0; i < 1000; i++) yield return new object[] { i };
}

// instead of just [Fact]
[Theory]
[MemberData("Iterations")]
public void AcceptAsyncV6BoundToAnyV6_Success(int iter)
{
    ...
}
```
so that when I run the tests, the particular test in question gets run repeatedly as many times as I like.  This doesn't of course simulate all the relevant conditions, e.g. if other tests were running in parallel to trigger the failure, but just wanted to mention it.

**Comment by eerhardt** [comment]
I haven't been able to repro this failure using the above procedure, and we haven't seen this issue in the CI lately. It more than likely was fixed with other changes in the area.

Closing. We can reactivate it if we start seeing it again.

**Comment by stephentoub** [comment]
This just happened again:

**Comment by ericeil** [comment]
This looks a lot like a use-after-free problem (getting the endpoint from a subsequently-allocated Socket).  This may be fixed by #6126.

**Comment by stephentoub** [comment]
Failed again here:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/osx_debug_tst_prtest/2209/console
```
12:39:43    System.Net.Sockets.Tests.DualMode.AcceptV6BoundToAnyV6_Success [FAIL]
12:39:43       System.ArgumentException : The supplied System.Net.Internals.SocketAddress is an invalid size for the System.Net.IPEndPoint end point.
12:39:43       Parameter name: socketAddress
12:39:43       Stack Trace:
12:39:43             at System.Net.Sockets.IPEndPointExtensions.Create(EndPoint thisObj, SocketAddress socketAddress)
12:39:43             at System.Net.Sockets.Socket.Accept()
12:39:43             at System.Net.Sockets.Tests.DualMode.Accept_Helper(IPAddress listenOn, IPAddress connectTo)
12:39:43             at System.Net.Sockets.Tests.DualMode.AcceptV6BoundToAnyV6_Success()
```

**Comment by ericeil** [comment]
This hasn't repro'd in more than a month, and I can't reproduce it, even with GCStress.  It's likely this was already fixed, possibly by dotnet/coreclr#4266.
